tensorrt_llm/_torch/pyexecutor/sampler/sampler.py

```
@@ -3971,6 +4177,76 @@ def provision_bias_index() -> int:
+            per_step_slots_cuda = seq_slots_cuda[first_req : last_req + 1]
+        else:
+            group_seq_slots = seq_slots[group_req_indices].tolist()
+            group_steps = req_num_steps[group_req_indices].tolist()
+            # Build a per-STEP slot layout aligned with group_strategies_per_step
+            # (each request contributes req_num_steps rows). The per-row decay mask is
+            # not built here: it is gathered on-device from the is_decay_slot gate
+            # below. A decay-active row must be single-token; reject only if a *decay*
+            # row is multi-step (a non-decay draft row in the same group is fine).
+            per_step_slots: list[int] = []
+            has_decay = False
+            for slot, steps in zip(group_seq_slots, group_steps):
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)